### PR TITLE
 Fix write_pandas() truncate-only flow to prevent CREATE TABLE IF NOT EXISTS

### PR DESCRIPTION
**Pull request by sfc-gh-kterada**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

Addresses [SNOW-1184290](https://snowflakecomputing.atlassian.net/browse/SNOW-1184290)

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

Builds on the changes committed in SNOW-733213 to enforce truncate-only behavior by excluding CREATE TABLE IF NOT EXISTS when auto_create_table = False and overwrite = True.

[SNOW-1184290]: https://snowflakecomputing.atlassian.net/browse/SNOW-1184290?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by sfc-gh-stan — APPROVED** [review]

File: src/snowflake/connector/pandas_tools.py [hunk 1]
```diff
@@ -393,13 +393,14 @@ def drop_object(name: str, object_type: str) -> None:
...
         )
 
-        create_table_sql = (
-            f"CREATE {table_type.upper()} TABLE IF NOT EXISTS {target_table_location} "
-            f"({create_table_columns})"
-            f" /* Python:snowflake.connector.pandas_tools.write_pandas() */ "
-        )
-        logger.debug(f"auto creating table with '{create_table_sql}'")
-        cursor.execute(create_table_sql, _is_internal=True)
+        if auto_create_table:
```
Comment (sfc-gh-kterada) [inline src/snowflake/connector/pandas_tools.py:396]: I have read the CLA Document and I hereby sign the CLA

**Review by sfc-gh-kterada — COMMENTED** [review]
I have read the CLA Document and I hereby sign the CLA

**Comment by sfc-gh-stan** [comment]
recheck

**Review by sfc-gh-kterada — COMMENTED** [review]
recheck

File: DESCRIPTION.md
Comment (sfc-gh-kterada) [inline DESCRIPTION.md:1]: I have read the CLA Document and I hereby sign the CLA

**Comment by sfc-gh-kterada** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by sfc-gh-aling** [comment]
although this PR fixes the bug, however this is a breaking change that breaks the case in which table_name is specified, but auto_create_table is not set.

for example it breaks the test in snowpark python lib, to fix the breaking change we need to 
 update the tests: https://github.com/snowflakedb/snowpark-python/pull/1371/files

we have yanked in 3.8.0 release and will do a 3.8.1 release without this commit.
we need to rethink about doing the fix in a non-breaking way.

**Pull request merged**